### PR TITLE
Add Shikimori provider

### PR DESCRIPTION
**Pull request by M9snikFeed**

### CODE REVIEW

**Comment by atymic** [comment]
Can you describe what this provider does?

**Comment by M9snikFeed** [comment]
Authorizes the user through Oauth2 on the Shikimori.one website. (online anime library). Can receive name, nickname, id, refresh and access tokens.

File: src/Shikimori/composer.lock [hunk 1]
```diff
@@ -0,0 +1,3066 @@
+{
```
Comment (lucasmichot) [inline src/Shikimori/composer.lock:1]: composer.lock should not have been commited.
Can you remove it @M9snikFeed ?

File: src/Shikimori/Provider.php [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+    public const IDENTIFIER = 'SHIKIMORI';
+
+    /**
+     * {@inheritdoc}
+     */
+    protected $scopes = [
+        'user_rates'
+    ];
+
+    protected $scopeSeparator = ' ';
```
Comment (lucasmichot) [inline src/Shikimori/Provider.php:24]: Missing `inheritdoc` docblock.

File: src/Shikimori/Provider.php [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+
+    protected function getShikimoriUrl()
```
Comment (lucasmichot) [inline src/Shikimori/Provider.php:26]: missing docblock

File: src/Shikimori/Provider.php [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+    {
+        return $this->getConfig('base_url');
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public static function additionalConfigKeys()
+    {
+        return ['base_url'];
```
Comment (lucasmichot) [inline src/Shikimori/Provider.php:36]: Are you really sure about this ?
By reading your docs it seems that `base_url` will always be `https://shikimori.one`

**Review by lucasmichot — CHANGES_REQUESTED** [review]
Some changes are required @M9snikFeed
Also can you add a `README` file?

File: src/Shikimori/Provider.php [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+    /**
+     * {@inheritdoc}
+     */
+    protected function mapUserToObject(array $user)
+    {
+        return (new User())->setRaw($user)->map([
+            'id'       => $user['id'],
+            'nickname' => $user['nickname'],
+            'name'     => $user['name'],
+            'sex'      => $user['sex'],
```
Comment (lucasmichot) [inline src/Shikimori/Provider.php:78]: If shikimori.one service is still being worked on
This field could also be renamed to `gender`

File: src/Shikimori/composer.lock (hunk 1 above)
Comment (M9snikFeed) [reply src/Shikimori/composer.lock:1]: Yes

**Review by M9snikFeed — COMMENTED** [review]
corrected

**Comment by lucasmichot** [comment]
> corrected

Thanks @M9snikFeed, could you also check the other comments in the review ?

**Comment by M9snikFeed** [comment]
> > corrected
> 
> Thanks @M9snikFeed, could you also check the other comments in the review ?

I looked at everything and fixed it

File: src/Shikimori/Provider.php (hunk 4 above)
Comment (atymic) [reply src/Shikimori/Provider.php:41]: @M9snikFeed

File: src/Shikimori/composer.lock (hunk 1 above)
Comment (atymic) [reply src/Shikimori/composer.lock:1]: You didn't remove it

**Review by atymic — CHANGES_REQUESTED** [review]
Please also add to the monorepobuilder.yml split script

**Comment by lucasmichot** [comment]
Also can you remove `composer.lock`?

**Comment by atymic** [comment]
@M9snikFeed

**Comment by atymic** [comment]
@M9snikFeed

**Comment by M9snikFeed** [comment]
> @M9snikFeed 

Sorry, I just got to the hospital with covid.  As soon as I leave, I will correct everything)

**Comment by atymic** [comment]
> > @M9snikFeed
> 
> Sorry, I just got to the hospital with covid. As soon as I leave, I will correct everything)

Wow my apologies, I hope you get better soon and no rush! Rest up.

**Pull request closed without merging**